File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        
        vector<int>ds;
        vector<vector<int>>ans;
        int k =1;
        
        while(numRows--){
            
            vector<int>ds(k);
            ds[0] =1;
            ds[k-1] = 1;
            
            for(int j =1;j<k-1;j++){
                ds[j] = ans.back()[j-1] + ans.back()[j];
            }
            
            ans.push_back(ds);
            k++;
            
        }
        
        return ans;
        
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
def generaterow(row):

    li= [1]
    if row == 1: return li
    temp =1
    for i in range(1,row-1):
        temp = temp * (row - i)
        temp = temp//i
        li.append(temp)

    li.append(1)
    return li

class Solution:
            
    def generate(self, numRows: int) -> List[List[int]]:
        ans =[]
        for i in range(1, numRows+1): ans.append(generaterow(i))
        return ans
        